### PR TITLE
LS and monaco commands integration

### PR DESCRIPTION
**Pull request by akosyakov**
Our command registry is treated as a main source of commands. LS can contribute commands to it. Monaco is hooked up with it via an implementation of `monaco.commands.ICommandService`.

### CODE REVIEW

**Comment by akosyakov** [comment]
Should fix https://github.com/theia-ide/theia/issues/76

File: src/application/common/command.ts [hunk 1]
```diff
@@ -15,9 +15,9 @@ export interface Command {
     iconClass?: string;
 }
 export interface CommandHandler {
-    execute(): any;
-    isEnabled(): boolean;
-    isVisible?(): boolean;
+    execute(...args: any[]): any;
+    isEnabled(...args: any[]): boolean;
```
Comment (svenefftinge) [inline src/application/common/command.ts]: I'd rather make `isEnabled` optional and then not overload the functions below with the additional function signature.

File: src/application/common/command.ts [hunk 2]
```diff
@@ -39,12 +44,36 @@ export class CommandRegistry {
...
-        for (let contrib of contributions) {
+        for (const contrib of contributions) {
             contrib.contribute(this);
         }
     }
 
-    registerCommand(command: Command): Disposable {
+    registerCommand(command: string | Command): Disposable;
+    registerCommand(command: string | Command, handler: CommandHandler): Disposable;
+    registerCommand(command: string | Command, handler: (...args: any[]) => any, thisArg?: any): Disposable;
```
Comment (svenefftinge) [inline src/application/common/command.ts]: why do we need the this binding?

File: src/application/common/command.ts (hunk 1 above)
Comment (akosyakov) [inline src/application/common/command.ts]: agreed with optional, but with overloads, an using side looks better in simple cases, e.g.
```
commands.registerCommand(APPLY_WORKSPACE_EDIT, (changes: WorkspaceEdit) =>
    !!this.workspace.applyEdit && this.workspace.applyEdit(changes)
)
```
against
```
commands.registerCommand({ id: APPLY_WORKSPACE_EDIT }, {
    execute: (changes: WorkspaceEdit) =>
        !!this.workspace.applyEdit && this.workspace.applyEdit(changes)
})
```

**Comment by akosyakov** [comment]
@svenefftinge good to merge?

**Pull request merged**